src/main/java/duke/task/DatedTask.java

```diff
@@ -0,0 +1,42 @@
+package duke.task;
+
+import duke.DukeException;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Encapsulates a Task that is associated with a date.
+ */
+public abstract class DatedTask extends Task {
```

Comment:
Personally did not think of creating another dated task class for event and deadline class to inherit from. Good application of OOP concepts! 👍